src/components/EmojiAvatar/EmojiAvatar.tsx

```diff
@@ -0,0 +1,86 @@
+/* External dependencies */
+import React, {
+  useCallback,
+  useMemo,
+} from 'react'
+import { isEmpty, isNil } from 'lodash-es'
+
+/* Internal dependencies */
+import { Status } from 'Components/Status'
+import EmojiAvatarProps, { EmojiAvatarSize } from './EmojiAvatar.types'
+import {
+  StatusWrapper,
+  Wrapper,
+  EmojiIcon,
+} from './EmojiAvatar.styled'
+
+export const EMOJI_AVATAR_TEST_ID = 'bezier-react-emoji-avatar'
+
+function EmojiAvatar({
+  as,
+  style,
+  emojiUrl,
+  className,
+  interpolation,
+  status,
+  size = EmojiAvatarSize.Size24,
+  testId = EMOJI_AVATAR_TEST_ID,
+  onClick,
+  children,
+}: EmojiAvatarProps) {
+  const handleClickIcon = useCallback((event: React.MouseEvent<HTMLDivElement>) => {
+    if (!isNil(onClick)) {
+      onClick(event)
+    }
+  }, [onClick])
+
+  const StatusElement = useMemo(() => {
+    if (
+      (isEmpty(children) && !status)
+      || (children && !React.isValidElement(children))
+    ) {
+      return null
+    }
+
+    const Contents = (() => {
+      if (children) { return children }
+      if (status) {
+        return (
+          <Status type={status} />
+        )
+      }
+      return null
+    })()
+
+    return (
+      <StatusWrapper>
+        { Contents }
+      </StatusWrapper>
+    )
+  }, [
+    status,
+    children,
+  ])
+
+  return (
+    <Wrapper
+      as={as || !isNil(onClick) ? 'button' : 'div'}
```

Comment:
operator precedence에 따라서 `||`가 먼저 evaluate됩니다.
```suggestion
      as={as || (!isNil(onClick) ? 'button' : 'div')}
```

Reply:
요거 이제 관련 사항은 필요 없어졌지만.. 이런 부분들이 테스트가 미리 작성되어있었다면 놓치지 않았겠다는 아쉬움이 드네요.